### PR TITLE
IPv4 only support for some UDP features at TizenRT

### PR DESCRIPTION
**Pull request by pmarcinkiew**
Several UDP features were enabled for IPv4 only implementations in TizenRT.

Enabled functions:
uv_udp_set_ttl, uv_udp_set_multicast_ttl, uv_udp_set_multicast_loop, uv_udp_set_multicast_interface

### CODE REVIEW

**Comment by pmarcinkiew** [comment]
See:

https://github.com/Samsung/libtuv/issues/89

**Comment by LaszloLango** [comment]
Please fix the DCO in the commit message.

**Comment by yichoi** [comment]
@pmarcinkiew rebase please

**Comment by pmarcinkiew** [comment]
Could you merge this PR?

File: src/unix/udp.c [hunk 1]
```diff
@@ -797,6 +797,38 @@ int uv_udp_set_broadcast(uv_udp_t* handle, int on) {
...
+ * These flags are not supported for IPv6 by TizenRT
+ * The uv_set_socckopt expects both IPv6 and IPv4 versions of flag and checks
+ * internally, which flags should be used based on address family.
+ * This solution allows compilation of calls for TizenRT.
+ * An error for not supported flag will occur at runtime instead
+ * of compilation. It is better to be able to set TTL at least one protocol
+ * than not being able to set TTL at all for.
+ */
+static int get_ipv6_unicast_hops() {
+    return -1;
```
Comment (LaszloLango) [inline src/unix/udp.c]: wrong indentation

File: src/unix/udp.c [hunk 2]
```diff
@@ -797,6 +797,38 @@ int uv_udp_set_broadcast(uv_udp_t* handle, int on) {
...
+}
+static int get_ipv6_multicast_hops() {
+    return -1;
```
Comment (LaszloLango) [inline src/unix/udp.c]: ditto

File: src/unix/udp.c [hunk 3]
```diff
@@ -797,6 +797,38 @@ int uv_udp_set_broadcast(uv_udp_t* handle, int on) {
...
+}
+static int get_ipv6_multicast_loop() {
+    return -1;
```
Comment (LaszloLango) [inline src/unix/udp.c]: ditto

File: src/unix/udp.c [hunk 4]
```diff
@@ -797,6 +797,38 @@ int uv_udp_set_broadcast(uv_udp_t* handle, int on) {
...
+}
+#else
+static int get_ipv6_unicast_hops() {
+    return IPV6_UNICAST_HOPS;
```
Comment (LaszloLango) [inline src/unix/udp.c]: ditto

File: src/unix/udp.c [hunk 5]
```diff
@@ -797,6 +797,38 @@ int uv_udp_set_broadcast(uv_udp_t* handle, int on) {
...
+}
+static int get_ipv6_multicast_hops() {
+    return IPV6_MULTICAST_HOPS;
```
Comment (LaszloLango) [inline src/unix/udp.c]: ditto

File: src/unix/udp.c [hunk 6]
```diff
@@ -797,6 +797,38 @@ int uv_udp_set_broadcast(uv_udp_t* handle, int on) {
...
+}
+static int get_ipv6_multicast_loop() {
+    return IPV6_MULTICAST_LOOP;
```
Comment (LaszloLango) [inline src/unix/udp.c]: ditto

File: src/unix/udp.c [hunk 7]
```diff
@@ -916,14 +950,17 @@ int uv_udp_set_multicast_interface(uv_udp_t* handle, const char* interface_addr)
...
   } else {
     assert(0 && "unexpected address family");
     abort();
   }
 
   return 0;
 }
-#endif
+
+
```
Comment (LaszloLango) [inline src/unix/udp.c]: Unnecessary empty lines

File: src/unix/udp.c [hunk 8]
```diff
@@ -797,6 +797,38 @@ int uv_udp_set_broadcast(uv_udp_t* handle, int on) {
...
+static int get_ipv6_unicast_hops() {
+    return IPV6_UNICAST_HOPS;
+}
+static int get_ipv6_multicast_hops() {
+    return IPV6_MULTICAST_HOPS;
+}
+static int get_ipv6_multicast_loop() {
+    return IPV6_MULTICAST_LOOP;
+}
+#endif
```
Comment (LaszloLango) [inline src/unix/udp.c]: `#endif /* defined(__TIZENRT__) */`

File: src/unix/udp.c (hunk 1 above)
Comment (pmarcinkiew) [reply src/unix/udp.c]: done

File: src/unix/udp.c (hunk 2 above)
Comment (pmarcinkiew) [reply src/unix/udp.c]: done

File: src/unix/udp.c (hunk 3 above)
Comment (pmarcinkiew) [reply src/unix/udp.c]: done

File: src/unix/udp.c (hunk 4 above)
Comment (pmarcinkiew) [reply src/unix/udp.c]: done

File: src/unix/udp.c (hunk 5 above)
Comment (pmarcinkiew) [reply src/unix/udp.c]: done

File: src/unix/udp.c (hunk 6 above)
Comment (pmarcinkiew) [reply src/unix/udp.c]: done

File: src/unix/udp.c (hunk 7 above)
Comment (pmarcinkiew) [reply src/unix/udp.c]: done

File: src/unix/udp.c [hunk 9]
```diff
@@ -797,6 +797,37 @@ int uv_udp_set_broadcast(uv_udp_t* handle, int on) {
...
+}
+static int get_ipv6_multicast_hops() {
+  return -1;
+}
+static int get_ipv6_multicast_loop() {
+  return -1;
+}
+#else
+static int get_ipv6_unicast_hops() {
+  return IPV6_UNICAST_HOPS;
```
Comment (LaszloLango) [inline src/unix/udp.c]: https://github.com/Samsung/libtuv/pull/93 introduced these macros. Wouldn't be better to change them to `-1` at line 72:80?

**Comment by pmarcinkiew** [comment]
In my opinion https://github.com/Samsung/libtuv/pull/93 solves issue I was investigating. Let's close this PR.

**Pull request closed without merging**